src/sql/src/plan/statement/ddl.rs

```diff
@@ -1110,6 +1111,24 @@ fn kafka_sink_builder(
         bail!("exactly-once requires a consistency topic");
     }
 
+    if include_consistency {
+        for item in root_dependencies.iter() {
+            if item.item_type() == CatalogItemType::Source
+                && !item.source_connector()?.yields_stable_input()
+            {
+                bail!(
+                    "all input sources of a consistent Kafka sink must be replayable, {} is not",
```

Comment:
why don't we just say

"all input sources to a consistent Kafka sink must be Kafka sources"

I know this isn't exactly true right now but it will be true before users can use it?

Reply:
I suspect out in the field other exotic sources may need to be brought in to get some query/sink to run by any means necessary:
- a VALUES view
- a File source that has been declared using some future option to be forver present and immutable.